build/utils/yarn.go

```diff
@@ -94,11 +94,11 @@ func GetYarnExecutable() (string, error) {
 	return yarnExecPath, nil
 }
 
-// GetYarnDependencies returns a map of the dependencies of a Yarn project and the root package of the project.
+// Returns a map of the dependencies of a Yarn project and the root package of the project.
 // The keys are the packages' values (Yarn's full identifiers of the packages), for example: '@scope/package-name@1.0.0'
 // (for yarn v < 2.0.0) or @scope/package-name@npm:1.0.0 (for yarn v >= 2.0.0).
 // Pay attention that a package's value won't necessarily contain its version. Use the version in package's details instead.
-func GetYarnDependencies(executablePath, srcPath string, packageInfo *PackageInfo, log utils.Log) (dependenciesMap map[string]*YarnDependency, root *YarnDependency, err error) {
+func GetYarnDependencies(executablePath, srcPath string, packageInfo *PackageInfo, log utils.Log, allowPartialResults bool) (dependenciesMap map[string]*YarnDependency, root *YarnDependency, err error) {
```

Comment:
Let's add to the function's description, the description for the function's arguments, and specifically the new `allowPartialResults` argument.
The same goes for the `buildYarnV1DependencyMap` function below.